apps/emqx_resource/src/emqx_resource_buffer_worker.erl

```diff
@@ -1441,7 +1422,11 @@ ack_inflight(InflightTID, Ref, Id, Index) ->
             [] ->
                 {0, false}
         end,
-    ok = dec_inflight_remove(InflightTID, Count, Removed),
+    FlushCheck = dec_inflight_remove(InflightTID, Count, Removed),
+    case FlushCheck of
+        continue -> ok;
+        flush -> ?MODULE:flush_worker(WorkerPid)
```

Comment:
Nit: not quite clear how `continue` opposes to `flush` 😅

Reply:
Would `no_flush` or `dont_flush` be better?  :thinking:

Reply:
Fixed here: https://github.com/emqx/emqx/pull/10740